File: Resueltos_por_competidor/Jacobo_Segura/Sereja_And_Dima.java
/* 
Codeforces 381A - Sereja and Dima
https://codeforces.com/problemset/problem/381/A
*/
import java.util.ArrayList;
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        ArrayList<Integer> g = new ArrayList<>();
        for(int i = 0; i<n; i++) {
        	g.add(sc.nextInt());
        }
        int s = 0;
        int d = 0;
        for(int i = 0; i<n; i++) {
        	if(i%2 == 0) {
        		if(g.get(0) > g.get(g.size() - 1)) {
            		s += g.get(0);
            		g.remove(0);
            	}else {
            		s += g.get(g.size() - 1);
            		g.remove(g.size() - 1);
            	}
        	}else {
        		if(g.get(0) > g.get(g.size() - 1)) {
            		d += g.get(0);
            		g.remove(0);
            	}else {
            		d += g.get(g.size() - 1);
            		g.remove(g.size() - 1);
            	}
        	}
        }
        System.out.println(s + " " + d);
    }
}

File: Resueltos_por_competidor/Jacobo_Segura/Sereja_And_Dima.cpp
/* 
Codeforces 381A - Sereja and Dima
https://codeforces.com/problemset/problem/381/A
*/
#include <bits/stdc++.h>
using namespace std;

int main() {
	int t;
	cin>>t;
	int* n = new (std::nothrow) int[t];
	
	for(int i = 0; i<t; i++){
	    cin>>n[i];
	}
	
	int izq = 0, der = t-1;
	int s = 0, d = 0;
	bool turnoSereja = true;
	
	while(izq <= der){
	    int g;
	    
	    if(n[izq] > n[der]){
	        g = n[izq];
	        izq++;
	    }else{
	        g = n[der];
	        der--;
	    }
	    
	    if(turnoSereja){
	        s += g;
	    }else{
	        d += g;
	    }
	    turnoSereja = !turnoSereja;
	}
	cout<<s<<" "<<d<<endl;
	return 0;
}
